tests/e2e/testsLibrary/CodeExecutionTests.ts

```diff
@@ -30,6 +33,24 @@ export function runTaskWithDialogShellAndOpenLink(taskName: string, expectedDial
     });
 }
 
+export function runTaskWithDialogShellDjangoWorkaround(taskName: string, expectedDialogText: string, urlSubPath: string, timeout: number) {
+    test(`Run command '${taskName}' expecting dialog shell`, async () => {
+        await topMenu.runTask(taskName);
+        await dialogWindow.waitDialog(timeout, expectedDialogText);
+        const dialogRedirectUrl: string = await dialogWindow.getApplicationUrlFromDialog(expectedDialogText);
+        const augmentedPreviewUrl: string = dialogRedirectUrl + urlSubPath;
+        await dialogWindow.closeDialog();
+        await dialogWindow.waitDialogDissappearance();
+        await driverHelper.getDriver().wait(async () => {
+            try {
+                const agent = new https.Agent({ rejectUnauthorized: false });
```

Comment:
Looks good just several questions:
 - Could you explain will the application url be in "https" mode in all cases?
 - It might be great to have the possibility to make "https" calls in some common place?

Reply:
* It should probably always be https as we want to run che deployments in the `--tls` mode with `--self-signed-cert` flags.
* I don't plan to keep it as a permanent solution. I have already opened an issue about the preview widget not working as intended in upstream ([link](https://github.com/eclipse/che/issues/15053)). once it's fixed, I will remove the https call and rewrite the test to use UI.

Reply:
- Does it possible variant if deploy che without tls support then this test will not work?
- It would be nice to create an issue (task) for removing temporary solution and link it with issue which describes a related problem. It will allow don't forget to remove workaround when bug will be fixed.

Reply:
* will the axios fail using http link if it has `httpsAcent` set explicitely? Shouldn't it still create an instance for `httpAgent` automatically?
* Will do :+1:

Reply:
- I am not sure about axios behavior, but if https agent works properly with http links I agree with you, that this test doesn't require additional updates.

Reply:
will check and come back to you.

Reply:
ok, it should be fine. the `httpAgent` is inherited from the default Agent config and shouldn't interfere.